webapp/channels/src/utils/utils.tsx

```diff
@@ -265,6 +281,36 @@ export function getRemainingDaysFromFutureTimestamp(timestamp?: number): number
     return Math.floor((utcFuture - utcToday) / MS_PER_DAY);
 }
 
+export function addTimeToTimestamp(timestamp: number, type: TimeUnit, diff: number, timeline: TimeDirection) {
+    switch (type) {
+    case TimeInformation.MILLISECONDS:
+        return timeline === TimeInformation.FUTURE ? timestamp + diff : timestamp - diff;
+    case TimeInformation.SECONDS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_SECOND) : timestamp + (diff * MS_PER_SECOND);
+    case TimeInformation.MINUTES:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_MINUTE) : timestamp + (diff * MS_PER_MINUTE);
+    case TimeInformation.HOURS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_HOUR) : timestamp + (diff * MS_PER_HOUR);
+    case TimeInformation.DAYS:
+        return timeline === TimeInformation.FUTURE ? timestamp + (diff * MS_PER_DAY) : timestamp + (diff * MS_PER_DAY);
```

Comment:
Should not the falsy expression be subtracting from the timestamp (for Seconds, Minutes, Hours and Days)?

Reply:
nits:
- Whether it's a Future timeline or not could be calculated outside the switch (ie. `isFuture`).
- Maybe the amount to be added/subtracted can be calculated outside the ternary.
- Milliseconds seems to be the default case, could be set along with the `default`.